Support Kitty's Unicode placeholders feature and support running loooooooongcat within tmux

haya14busa opened the pull request:
![Screenshot 2025-05-02 at 3 03 55](https://github.com/user-attachments/assets/5ebf44f3-ce84-46f9-8e17-4c84b112ba7f)
Tested with Ghostty terminal and tmux version=3.5a :cat::cat::cat::cat::cat:

Document: https://github.com/kovidgoyal/kitty/blob/04fbf63eabc8ec6120ca39ace3508c45384edf42/docs/graphics-protocol.rst#unicode-placeholders

haya14busa committed:
it worked!

haya14busa committed:
adjust cell size

haya14busa committed:
remove needless comments

yoichi commented:
After this change, with Terminal.app on macOS, longcat stops processing at

```
yoichi@Mac longcat % ./longcat 
Ga=q,i=22896,s=1,v=1,t=d,f=24;AAAA
```
If I press some key, then the image is displayed in pixmap mode as before.

mattn committed:
Reuse shared terminal query for kitty detection and cell size

- Route kitty graphics protocol and cell size queries through a
  QueryTerminal hook instead of duplicating terminal I/O.
- Drop the leaking reader goroutines and reads from os.Stdout.
- Restrict the graphics protocol probe to tmux to avoid mishandling
  on terminals such as Terminal.app.
- Query cell size only in Unicode placeholder mode.

mattn committed:
Merge branch 'master' into kitty-unicode-placeholder-and-tmux

mattn committed:
Update kitty graphics test for new NewEncoder signature
